src/renderer/components/item-object-list/item-list-layout.tsx

```diff
@@ -216,6 +231,38 @@ export class ItemListLayout extends React.Component<ItemListLayoutProps> {
     return this.applyFilters(filterItems, allItems);
   }
 
+  updateColumnFilter(checkboxValue: boolean, columnName: string) {
+    if (checkboxValue){
+      this.hiddenColumnNames.delete(columnName);
+    } else {
+      this.hiddenColumnNames.add(columnName);
+    }
+  }
+
+  showColumn(index: number): boolean {
+    const {renderTableHeader} = this.props;
+
+    if (!this.canBeConfigured) return true;
+
+    return !this.hiddenColumnNames.has(renderTableHeader[index].showWithColumn ?? renderTableHeader[index].className);
+  }
+
+  @computed get canBeConfigured(): boolean {
+    const { isConfigurable, tableId, renderTableHeader } = this.props;
+
+    if (!isConfigurable || !tableId) {
+      return false;
+    }
+
+    if (!renderTableHeader?.every(({ className }) => className)) {
```

Comment:
It cannot be `required` since that would require changing every table. I think that a new prop would probably be better but I didn't think I could justify the extra effort on the part of the developers.

Reply:
I thought about it but I've decided to avoid global changes in root objects. The current number of tables is not too huge and I can implement new prop for all tables and set the prop of every column.

Reply:
You could reuse `title` and add a new optional property `hideTitle?: boolean`. That way columns similar to `className: "warning"` could still have it not rendered in the table (but do render it in the menu)